exercises/parallel-letter-frequency/src/test/java/ParallelLetterFrequencyTest.java

```diff
@@ -0,0 +1,216 @@
+        b.append(OdeAnDieFreude);
+        b.append(Wilhelmus);
+        b.append(StarSpangledBanner);
+
+        ParallelLetterFrequency p = new ParallelLetterFrequency(b.toString());
+
+        assertEquals(new Integer(49), p.letterCounts().get((int) 'a'));
+        assertEquals(new Integer(56), p.letterCounts().get((int) 't'));
+        assertEquals(new Integer(2), p.letterCounts().get((int) 'ü'));
+    }
+
+    @Test
+    @Ignore
+    public void tenThreadPoolIsFasterThanSingleThreadPool()
+            throws InterruptedException, ExecutionException {
+
+        ForkJoinPool singleThreadPool = new ForkJoinPool(1);
+        ForkJoinPool tenThreadPool = new ForkJoinPool(10);
+
+        String input = "abcdefghijklmnopqrstuvwxyz";
+        StringBuilder b = new StringBuilder();
+        for (int i = 1; i < 10000000; i++) {
+            b.append(input);
+        }
+
+        Map<Integer, Integer> expectedOutput = new HashMap<Integer, Integer>() {
+            {
+                put((int) 'a', 10000000);
+                put((int) 'b', 10000000);
+                put((int) 'c', 10000000);
+                put((int) 'd', 10000000);
+                put((int) 'e', 10000000);
+                put((int) 'f', 10000000);
+                put((int) 'g', 10000000);
+                put((int) 'h', 10000000);
+                put((int) 'i', 10000000);
+                put((int) 'j', 10000000);
+                put((int) 'k', 10000000);
+                put((int) 'l', 10000000);
+                put((int) 'm', 10000000);
+                put((int) 'n', 10000000);
+                put((int) 'o', 10000000);
+                put((int) 'p', 10000000);
+                put((int) 'q', 10000000);
+                put((int) 'r', 10000000);
+                put((int) 's', 10000000);
+                put((int) 't', 10000000);
+                put((int) 'u', 10000000);
+                put((int) 'v', 10000000);
+                put((int) 'w', 10000000);
+                put((int) 'x', 10000000);
+                put((int) 'y', 10000000);
+                put((int) 'z', 10000000);
+            }
+        };
+
+        Long singleStartTime = System.currentTimeMillis();
+        Map<Integer, Integer> singleCounts = singleThreadPool.submit(
+                () -> (new ParallelLetterFrequency(b.toString()).letterCounts())
+        ).get();
+        Long singleJobTime = System.currentTimeMillis() - singleStartTime;
+
+        Long tenStartTime = System.currentTimeMillis();
+        Map<Integer, Integer> actualCounts = tenThreadPool.submit(
+                () -> (new ParallelLetterFrequency(b.toString()).letterCounts())
+        ).get();
+        Long tenJobTime = System.currentTimeMillis() - tenStartTime;
+
+        System.out.println("Single threaded computation took: " + singleJobTime + " ms.");
+        System.out.println("Multi threaded computation took: " + tenJobTime + " ms.");
+
+        assertTrue(singleJobTime - tenJobTime > 0);
```

Comment:
A non-parallelized implementation might well pass this test with non-trivial frequency. Would it make sense to either

1. run the computations many times for each pool, and average the results, or
2. assert something like `singleJobTime - 2 * tenJobTime > 0`?

Reply:
@stkent I had originally done it like (2)! The results can vary quite a bit, but I think 2x is safe.  It's usually been closer to 3x on my laptop.

Reply:
Sweet; let's go with `2 *` as a first attempt then?

Reply:
Yeah, um...on that: I did some testing this AM and was pretty consistently getting ~1.8x rather than two, which seems weird to me.  Not sure what's changed since the first pass, but I'm going to investigate a bit more.

Reply:
🤞

We could also drop to 1.5x?

Reply:
@stkent 1.5 seems to be working :).  I also made a change where the number of threads defaults to the number of CPUs allocated to the JVM.  Otherwise I think the performance of multiple threads might actually be worse in this case.